pallets/randomness/src/lib.rs

```diff
@@ -286,12 +290,13 @@ pub mod pallet {
 		}
 		pub(crate) fn concat_and_hash(a: T::Hash, b: H256, index: u8) -> Vec<[u8; 32]> {
 			let mut output: Vec<[u8; 32]> = Vec::new();
+			let mut s = Vec::new();
 			for i in 0u8..index {
-				let mut s = Vec::new();
 				s.extend_from_slice(a.as_ref());
 				s.extend_from_slice(b.as_ref());
 				s.extend_from_slice(&[i]);
-				output.push(sp_io::hashing::blake2_256(&s))
+				output.push(sp_io::hashing::blake2_256(&s));
+				s.clear();
```

Comment:
Confirmed `clear()` does not free the allocated space:

`Note that this method has no effect on the allocated capacity of the vector.`

:)